example/v5_no_tls_prop.cpp

```diff
@@ -115,10 +117,10 @@ void client_proc(Client& c) {
         mqtt::v5::property::topic_alias_maximum(0x1234U),
         mqtt::v5::property::request_response_information(true),
         mqtt::v5::property::request_problem_information(false),
-        mqtt::v5::property::user_property("key1", "val1"),
-        mqtt::v5::property::user_property("key2", "val2"),
-        mqtt::v5::property::authentication_method("test authentication method"),
-        mqtt::v5::property::authentication_data("test authentication data")
+        mqtt::v5::property::user_property("key1"_mb, "val1"_mb),
```

Comment:
This PR is already very big and contains many(?) breaking changes. I'd like to consider the issue after the PR merged. Could you create a new issue for that?

Reply:
I can make an issue.

I'm not sure it's a good idea.

Do you think it's a good idea to offer user-defined literals for all of the property types?

Reply:
I think that it is good idea for all single string-base properties such as Content Type, Response Topic. 
It is not good for binary-base properties such as Correlation Data.

UserProperty is difficult. I think that user defined literals is not sutable for UserProperty. Because separator is required.  I think that `mqtt::v5::property::user_property("key1"_mb, "val1"_mb)` is simple enough.